WooCommerce/Classes/Authentication/Navigation Exceptions/JetpackSetupWebViewController.swift

```diff
@@ -0,0 +1,135 @@
+        let indicator = UIActivityIndicatorView(style: .medium)
+        indicator.hidesWhenStopped = true
+        return indicator
+    }()
+
+    /// Strong reference for the subscription to update progress bar
+    private var progressSubscription: AnyCancellable?
+
+    init(siteURL: String, onCompletion: @escaping () -> Void) {
+        self.siteURL = siteURL
+        self.completionHandler = onCompletion
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureNavigationBar()
+        configureWebView()
+        configureProgressBar()
+        startLoading()
+    }
+}
+
+private extension JetpackSetupWebViewController {
+    func configureNavigationBar() {
+        title = Localization.title
+        navigationItem.rightBarButtonItem = UIBarButtonItem(customView: activityIndicator)
+    }
+
+    func configureWebView() {
+        view.addSubview(webView)
+        NSLayoutConstraint.activate([
+            view.leadingAnchor.constraint(equalTo: webView.leadingAnchor),
+            view.trailingAnchor.constraint(equalTo: webView.trailingAnchor),
+            view.safeTopAnchor.constraint(equalTo: webView.topAnchor),
+            view.bottomAnchor.constraint(equalTo: webView.bottomAnchor),
+        ])
+    }
+
+    func configureProgressBar() {
+        view.addSubview(progressBar)
+        NSLayoutConstraint.activate([
+            view.leadingAnchor.constraint(equalTo: progressBar.leadingAnchor),
+            view.trailingAnchor.constraint(equalTo: progressBar.trailingAnchor),
+            view.safeTopAnchor.constraint(equalTo: progressBar.topAnchor)
+        ])
+    }
+
+    func startLoading() {
+        guard let escapedSiteURL = siteURL.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed),
+              let url = URL(string: String(format: Constants.jetpackInstallString, escapedSiteURL, Constants.mobileRedirectURL)) else {
+            return
+        }
+        progressSubscription = webView.publisher(for: \.estimatedProgress)
+            .sink { [weak self] progress in
+                if progress == 1 {
+                    self?.progressBar.setProgress(0, animated: false)
+                } else {
+                    self?.progressBar.setProgress(Float(progress), animated: true)
+                }
+            }
+        let request = URLRequest(url: url)
+        webView.load(request)
+    }
+}
+
+extension JetpackSetupWebViewController: WKNavigationDelegate {
+    func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void) {
+        let navigationURL = navigationAction.request.url?.absoluteString
+        switch navigationURL {
+        case let .some(url) where url == Constants.mobileRedirectURL:
```

Comment:
Super nit: 
Would it make sense to add some comments about this step?

Some details like,
1. We expect the Jetpack installation to be complete when we navigate to this URL.
1. This URL is triggered by calypso.

Reply:
Also updated in a061e812fa5beb38eca5f1c699e40feb2ccb1a58.